Source/ZXing.Net.Mobile.WindowsUniversal/MobileBarcodeScanner.cs

```diff
@@ -3,53 +3,72 @@
 using Windows.UI.Core;
 using Windows.UI.Xaml;
 using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Navigation;
 
 namespace ZXing.Mobile
 {
     public class MobileBarcodeScanner : MobileBarcodeScannerBase
     {
         public MobileBarcodeScanner () : base ()
         {
-            //this.Dispatcher = Windows.Current.Dispatcher;
         }
 
         public MobileBarcodeScanner(CoreDispatcher dispatcher) : base()
         {
-            this.Dispatcher = dispatcher;
+            Dispatcher = dispatcher;
         }
 
         public CoreDispatcher Dispatcher { get; set; }
 
         public Frame RootFrame { get; set; }
 
+        private Frame CurrentFrame
+        {
+            get
+            {
+                var currentFrame = RootFrame ??
+                                   Window.Current.Content as Frame ??
+                                   ((FrameworkElement) Window.Current.Content).GetFirstChildOfType<Frame>();
+
+                var currentPage = currentFrame.Content as Page;
+                if (currentPage != null && currentPage.NavigationCacheMode == NavigationCacheMode.Disabled)
+                    Log("WARNING: you may need to set {0}=\"Enabled\" for your Page to solve problems with updating UI after scanning is completed",
```

Comment:
Logging now implemented via Debug.WriteLine so it's will be not compiled on release. We might use Console.WriteLine (is it works here?) or mby (especially for NavigationCacheMode and other critical moments) other mechanism.